docker-compose.image.example.yaml

```diff
@@ -1,75 +1,208 @@
-version: '3.8'
+version: "3.8"
 services:
   backend:
     image: "superagidev/superagi:main"
+    restart: unless-stopped
     depends_on:
-      - super__redis
-      - super__postgres
-    networks:
-      - super_network
-    env_file:
-      - config.yaml
-    command: ["/app/wait-for-it.sh", "super__postgres:5432","-t","60","--","/app/entrypoint.sh"]
+      redis:
+        condition: service_healthy
+      postgres:
+        condition: service_healthy
+    environment:
+      - PINECONE_API_KEY=$PINECONE_API_KEY
```

Comment:
is it necessary to keep all the keys in `environment`
since config.yaml will load all environment variables so we can use get_config

Reply:
I think it's best practice to have env vars listed so that you can differentiate them between containers for security purpose, also in this way you can easly override them via standard env definition in docker, like in:

`docker compose run -e DEBUG=1 web python console.py`

loading them via python/yaml makes multi env deployments harder

Reply:
We can anyway override it 
![env edit (1)](https://github.com/TransformerOptimus/SuperAGI/assets/121793617/7bfe9924-2353-44ad-810e-10040f311d89)

Reply:
If you are not using an `env_file`, sure you can, if you specify the `env_file` key in docker-compose file then the env file is required so you can't go cli only.

Besides, there are the security concerns.